tracer/src/Datadog.Trace/Agent/Transports/ApiWebRequest.cs

```diff
@@ -46,5 +48,71 @@ public async Task<IApiResponse> PostAsync(ArraySegment<byte> bytes, string conte
                 return new ApiWebResponse((HttpWebResponse)exception.Response);
             }
         }
+
+        public async Task<IApiResponse> PostAsync(params MultipartFormItem[] items)
+        {
+            if (items is null)
+            {
+                ThrowHelper.ThrowArgumentNullException(nameof(items));
+            }
+
+            Log.Debug<int>("Sending multipart form request with {Count} items.", items.Length);
+
+            var boundary = "---------------------------" + DateTime.Now.Ticks.ToString("x");
+
+            _request.Method = "POST";
+            _request.ContentType = "multipart/form-data; boundary=" + boundary;
+
+            using (var requestStream = await _request.GetRequestStreamAsync().ConfigureAwait(false))
+            {
+                // Write form request using the boundary
+
+                var boundaryBytes = Encoding.ASCII.GetBytes("\r\n--" + boundary + "\r\n");
+
+                // Write each MultipartFormItem
+                foreach (var item in items)
+                {
+                    await requestStream.WriteAsync(boundaryBytes, 0, boundaryBytes.Length).ConfigureAwait(false);
+                    byte[] headerBytes;
+                    if (item.FileName is null)
+                    {
+                        headerBytes = Encoding.UTF8.GetBytes(
```

Comment:
Just checking that the header value is definitely UTF8 and not ASCII as above? Not sure which is correct in the spec (and how to handle names etc that don't fit if required)

Reply:
changed to ASCII and add some validations in [84ecdc7](https://github.com/DataDog/dd-trace-dotnet/pull/2962/commits/84ecdc7a6638c4593e83777456e9370644539de4)